x-pack/legacy/plugins/siem/public/components/drag_and_drop/helpers.ts

```diff
@@ -227,3 +227,96 @@ export const IS_DRAGGING_CLASS_NAME = 'is-dragging';
 
 /** This class is added to the document body while timeline field dragging */
 export const IS_TIMELINE_FIELD_DRAGGING_CLASS_NAME = 'is-timeline-field-dragging';
+
+export const allowTopN = ({
+  browserField,
+  fieldName,
+}: {
+  browserField: Partial<BrowserField> | undefined;
+  fieldName: string;
+}): boolean => {
+  const isAggregatable = browserField?.aggregatable ?? false;
+  const fieldType = browserField?.type ?? '';
+  const isAllowedType = contains(fieldType, [
+    'boolean',
+    'geo-point',
+    'geo-shape',
+    'ip',
+    'keyword',
+    'number',
+    'numeric',
+    'string',
+  ]);
+  const isWhitelistedNonBrowserField = contains(fieldName, [
```

Comment:
Not for this PR but I think we should ask ECS to have it in their documentation so we do not have to that, maybe we can add a TODO